modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -467,4 +470,83 @@ getQuantity(const RankTwoTensorTempl<T> & tensor,
                  scalarOptions().getRawNames());
   }
 }
+
+template <typename T>
+T
+getCylindricalComponent(const RankTwoTensorTempl<T> & tensor,
+                        const MooseEnum & scalar_type,
+                        const Point & point1,
+                        const Point & point2,
+                        const Point & curr_point,
+                        Point & direction)
+{
+  switch (scalar_type)
+  {
+    case 0:
+      return axialStress(tensor, point1, point2, direction);
+    case 1:
+      return hoopStress(tensor, point1, point2, curr_point, direction);
+    case 2:
+      return radialStress(tensor, point1, point2, curr_point, direction);
+    default:
+      mooseError("RankTwoCylindricalComponent Error: Pass valid scalar type - " +
+                 cylindricalOptions().getRawNames());
+  }
+}
+
+template <typename T>
+T
+getScalarDirectionalComponent(const RankTwoTensorTempl<T> & tensor,
+                              const MooseEnum & scalar_type,
+                              Point & direction)
+{
+  switch (scalar_type)
+  {
+    case 0:
+      return maxPrincipal(tensor, direction);
```

Comment:
I think the principal stresses don't belong in this function. The principal stresses are considered invariants of the stress tensor because you don't supply a direction -- it comes out of the calculation. 

This method should have direction be `const`, while it would not be const for the function that computes principal stresses.